Source/WebCore/css/CSSCounterStyle.h

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2022 Apple Inc.  All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <wtf/Forward.h>
+#include <wtf/text/AtomString.h>
+
+namespace WebCore {
+
+class StyleRuleCounterStyle;
+
+// The keywords that can be used as values for the counter-style `system` descriptor.
+// https://www.w3.org/TR/css-counter-styles-3/#counter-style-system
+enum class CounterStyleSystem : uint8_t {
+    Cyclic,
+    Numeric,
+    Alphabetic,
+    Symbolic,
+    Additive,
+    Fixed,
+    Extends
+};
+
+enum class CounterStyleSpeakAs : uint8_t {
+    Auto,
+    Bullets,
+    Numbers,
+    Words,
+    SpellOut,
+    CounterStyleNameReference,
+};
+
+class CounterStyle : public RefCounted<CounterStyle>, public CanMakeWeakPtr<CounterStyle> {
+public:
+    using Name = AtomString;
+    using Symbol = String;
+    using Ranges = Vector<std::pair<int, int>>;
+    using AdditiveSymbols = Vector<std::pair<Symbol, unsigned>>;
+
+    struct Pad {
+        unsigned m_padMinimumLength = 0;
+        Symbol m_padSymbol = "-"_s;
+        bool operator==(const Pad& other) const = default;
```

Comment:
Unfortunately I think these still fail to compile in some supported linux configurations.